src/v/coproc/utils.h

```diff
@@ -0,0 +1,72 @@
+#pragma once
+
+#include "coproc/reference_window_consumer.hpp"
+#include "model/fundamental.h"
+#include "model/record_batch_reader.h"
+#include "raft/types.h"
+#include "storage/log.h"
+#include "storage/log_reader.h"
+
+#include <optional>
+namespace coproc {
+
+struct batch_info {
+    model::offset last;
+    std::size_t total_size_bytes;
+    model::record_batch_reader rbr;
+};
+
+inline ss::future<std::optional<batch_info>>
+extract_offset(model::record_batch_reader&& rbr) {
```

Comment:
Maybe rename as this does more then extract the offset, also handle the TODO below